source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,293 @@
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start set pluralization
+   :end-before: end set pluralization
+
+As a result, {+odm+} stores ``Rey`` model class documents in the ``reyes`` 
+collection.
+
+.. note:: BSON Document Structure
+
+   When {+odm+} stores a document in a database, it serializes the Ruby object
+   to a BSON document that has the following structure:
+
+   .. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+      :language: ruby
+      :start-after: start BSON model
+      :end-before: end BSON model
+
+Persistence Context Attributes
+------------------------------
+
+Every model class contains the following methods, which you can use to retrieve
+information about where {+odm+} persists the model:
+
+- ``client_name``: Retrieves the client name
+- ``database_name``: Retrieves the database name
+- ``collection_name``: Retrieves the collection name
+
+The following example shows how to retrieve and print the names of the client, 
+database, and collection where documents for the ``Band`` class are persisted:
+
+.. io-code-block::
+
+  .. input:: /includes/data-modeling/persistence-configuration.rb
+     :language: ruby
+     :start-after: start persistence context attributes
+     :end-before: end persistence context attributes
+
+  .. output::
+     :language: ruby
+     :visible: false
+
+     default
+
+     my_bands
+
+     bands
+
+Customize Your Persistence Configuration
+----------------------------------------
+
+{+odm+} provides both model-level and runtime options for customizing your 
+persistence configuration. The following sections describe these options.
+
+Model-Level Persistence Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Suppose you want to store your model's documents in a collection with a 
+different name than the pluralized form of the model class name.
+You can use the ``store_in`` macro to change the collection, database, or client
+where {+odm+} stores a model's documents. The following example shows how
+to use the ``store_in`` macro to store documents from the ``Person`` class in
+a collection called ``citizens`` in the ``other`` database within a client
+named ``analytics``:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in example
+   :end-before: end store_in example
+
+The ``store_in`` macro can also accept a lambda function. This is useful if you
+want to define a persistence context with values that cannot use a constant string. 
+
+You might want to use this pattern in a multi-tenant application,
+where multiple users share common access to an application. By using
+a lambda, you can define a persistence context based on the current thread so
+that users cannot access each others' data.
+
+The following example stores documents in a database determined by the 
+current thread:
```

Comment:
In the example that follows, the discriminator isn't the _thread_, it's a _thread-local variable_. It might be nit-picking, but I feel like the distinction is significant. 

```suggestion
a lambda, you can define a persistence context based on thread-local information
so that users cannot access each others' data.

The following example stores documents in a database determined by a 
thread-local variable:
```

Reply:
Changed! I still specified that it's a variable local to the current thread... let me know if this is incorrect

Reply:
Yeah, that's fine. "Thread-local" implies "local to the current thread". 👍